### ISSUE TITLE
can't run linter goanalysis_metalinter: goanalysis_metalinter: nilness: package "capnp" (isInitialPkg: true, needAnalyzeSource: true): internal error: unhandled type *ir.CompositeValue

### ISSUE THREAD

**Issue by Lekensteyn** [golangci/golangci-lint#3578]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc.).
- [X] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

Since upgrading from golangci-lint v1.50.0 with go 1.19.2 to golangci-lint v1.51.1 with go 1.20, it crashes on certain code bases.

It might have something to do with a https://github.com/dominikh/go-tools @dominikh

Reproducer:
```
$ git clone https://github.com/capnproto/go-capnproto2 && cd go-capnproto2
$ git checkout 663a000ef79c30e55bc2d67163b4ebe85149bb72
$ docker run --rm -v "$PWD:/v" -w /v golangci/golangci-lint:v1.51.1 golangci-lint run .
ERRO [runner] Panic: nilness: package "capnp" (isInitialPkg: true, needAnalyzeSource: true): internal error: unhandled type *ir.CompositeValue: goroutine 2561 [running]:
runtime/debug.Stack()
	runtime/debug/stack.go:24 +0x65
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyzeSafe.func1()
	github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner_action.go:102 +0x155
panic({0xf42840, 0xc00356d6c0})
	runtime/panic.go:884 +0x213
honnef.co/go/tools/analysis/facts/nilness.impl.func1({0x1327b08, 0xc00677dea0})
	honnef.co/go/tools@v0.4.0/analysis/facts/nilness/nilness.go:234 +0x834
honnef.co/go/tools/analysis/facts/nilness.impl(0xc0080e0d20, 0xc003531180, 0xc0040d1cb8)
	honnef.co/go/tools@v0.4.0/analysis/facts/nilness/nilness.go:241 +0x372
honnef.co/go/tools/analysis/facts/nilness.run(0xc0080e0d20)
	honnef.co/go/tools@v0.4.0/analysis/facts/nilness/nilness.go:66 +0x11a
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyze(0xc001450b90)
	github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner_action.go:188 +0x9df
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyzeSafe.func2()
	github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner_action.go:106 +0x1d
github.com/golangci/golangci-lint/pkg/timeutils.(*Stopwatch).TrackStage(0xc0012a1180, {0x10d33d9, 0x7}, 0xc000057748)
	github.com/golangci/golangci-lint/pkg/timeutils/stopwatch.go:111 +0x4a
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyzeSafe(0xc000343680?)
	github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner_action.go:105 +0x85
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).analyze.func2(0xc001450b90)
	github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner_loadingpackage.go:80 +0xb4
created by github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).analyze
	github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner_loadingpackage.go:75 +0x1eb
WARN [runner] Can't run linter goanalysis_metalinter: goanalysis_metalinter: nilness: package "capnp" (isInitialPkg: true, needAnalyzeSource: true): internal error: unhandled type *ir.CompositeValue
ERRO Running error: 1 error occurred:
	* can't run linter goanalysis_metalinter: goanalysis_metalinter: nilness: package "capnp" (isInitialPkg: true, needAnalyzeSource: true): internal error: unhandled type *ir.CompositeValue
```

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version v1.51.1 built from b87d2c19 on 2023-02-05T13:26:55Z
```

</details>

### Configuration file

(default - none)

### Go environment

Reproduced on macOS and Linux (`golangci/golangci-lint:v1.51.1` Docker image).

### Verbose output of running

N/A

### Code example or link to a public repository

Reproduce with:
```go
package main

import capnp "capnproto.org/go/capnp/v3"

type TunnelCheckResult struct{ capnp.Struct }

var _ = TunnelCheckResult{}
```
and
```
go mod init reproducer
go get capnproto.org/go/capnp/v3@v3.0.0-alpha.3
golangci-lint run .
```

Versions up to https://github.com/capnproto/go-capnproto2/commit/663a000ef79c30e55bc2d67163b4ebe85149bb72 are affected. Since the next merge commit with https://github.com/capnproto/go-capnproto2/commit/d68553cb21ad9357238f9155c8a955b0e1353924 on top of, the issue can no longer be reproduced.

**Comment by dominikh** [comment]
Can reproduce with Staticcheck proper.

**Comment by dominikh** [comment]
Fixed by https://github.com/dominikh/go-tools/commit/e682878d73785996e74d3beb8b19cf4a87f84dd2, will be part of a release soon.

**Comment by gunta** [comment]
Reproduced here. We hope this gets released.

**Comment by ldez** [comment]
Fixed by #3591
